### PR TITLE
testing: immutability constraints on Row, Cell, GetCell, and SetCell

### LINKED ISSUE #101: testing: Test mutability constraints on exported functions
Ref #100 : There should be a few simple unit tests that ensure the assumptions made when exporting functions. For example, that `SetCell` can only set nil cells, and that `Row` and `Col` create copies.

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #101. A bit overkill I guess, but doesn't hurt to have them?

@adlerjohn, Should `eds.RowRoots` and `eds.ColRoots` also return copies? If so I would add a test for those still!

### CODE REVIEW

**Comment by distractedm1nd** [comment]
going to fail the new test in CI because RowRoots and ColRoots are mutable, will do in separate PR

**Comment by distractedm1nd** [comment]
Agreed, will do

**Comment by adlerjohn** [comment]
Converting to draft until ready for re-review. When ready please smash that "Ready for review" button!

**Comment by distractedm1nd** [comment]
@adlerjohn Decided to just add the fix for `RowRoots` and `ColRoots` in this branch, because it uses the same snippet as `Row` and `Col` anyways, so that it passes the new test again - this okay?

**Review by adlerjohn — APPROVED** [review]

**Comment by distractedm1nd** [comment]
@adlerjohn Is this waiting on another review? I don't have merge permissions here :)

**Review by liamsi — APPROVED** [review]

**Pull request merged**
